website/moneybirdsynchronization/services.py

```diff
@@ -151,90 +216,183 @@ def delete_external_invoice(obj):
+                send_sync_error(e, instance)
+                logging.exception("Moneybird synchronization error: %s", e)
 
-    response = moneybird.update_financial_statement(
-        financial_statement_id,
-        {
-            "financial_statement": {
-                "financial_mutations_attributes": {
-                    str(index_nr): moneybird_payment.to_moneybird()
-                }
-            }
-        },
-    )
-    moneybird_payment.moneybird_financial_statement_id = financial_statement_id
-    moneybird_payment.moneybird_financial_mutation_id = response["financial_mutations"][
-        index_nr
-    ]["id"]
-    moneybird_payment.save()
-    return moneybird_payment
 
+def sync_moneybird_payments():
+    """Create financial statements with all payments that haven't been synced yet.
 
-def create_moneybird_payment(payment):
-    """Create a payment on Moneybird."""
+    This creates one statement per payment type for which there are new payments.
+    """
     if not settings.MONEYBIRD_SYNC_ENABLED:
-        return None
+        return
 
-    financial_account_id = financial_account_id_for_payment_type(payment.type)
-    if financial_account_id is None:
-        # Don't sync if there's no financial account connected
-        # for example, 'other' payments (like wire transfers), don't have concrexit as source for financial
-        # mutation statements. So we ignore those and leave it to Moneybird itself.
-        return None
+    moneybird = get_moneybird_api_service()
 
-    moneybird_payment, _ = MoneybirdPayment.objects.get_or_create(payment=payment)
+    for payment_type in [Payment.TPAY, Payment.CARD, Payment.CASH]:
+        payments = Payment.objects.filter(
+            type=payment_type,
+            moneybird_payment__isnull=True,
+            created_at__date__gte=settings.MONEYBIRD_START_DATE,
+        )
 
-    if moneybird_payment.moneybird_financial_mutation_id is None:
-        last_payment_in_same_statement = MoneybirdPayment.objects.filter(
-            payment__type=payment.type,
-            payment__created_at__date=payment.created_at.date(),
-            moneybird_financial_statement_id__isnull=False,
-        ).first()
+        if payments.count() == 0:
+            continue
 
-        if last_payment_in_same_statement is not None:
-            # There already exists a statement we can add this payment to
-            existing_statement_id = (
-                last_payment_in_same_statement.moneybird_financial_statement_id
-            )
-            _add_payment_to_statement(moneybird_payment, existing_statement_id)
-        else:
-            _create_statement_and_add_payment(moneybird_payment, financial_account_id)
+        logger.info(
+            "Pushing %d %s payments to Moneybird.", payments.count(), payment_type
+        )
+
+        financial_account_id = financial_account_id_for_payment_type(payment_type)
+        reference = f"{payment_type} payments at {timezone.now():'%Y-%m-%d %H:%M'}"
```

Comment:
%H and %M can be left out

Reply:
Yeah, the thing is it might be confusing that in the 2023-10-15 statement, there would mainly be payments from 2023-10-16

Reply:
Hmm what about really just selecting the previous day? so 

`statement_day = timezone.now()-timedelta(days=1)`

and filter on

`created_at__date=statement_day`

Sounds more robust

Reply:
That's also nice but doesn't automatically fix stuff if the sync doesn't run on some day, or if it runs twice somehow (it'd make the same reference twice)